eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1879,25 +1883,24 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
         case _ => ()
       }
 
-      val previousChannelUpdate_opt = stateData match {
-        case data: DATA_NORMAL => Some(data.channelUpdate)
-        case _ => None
-      }
-
       (state, nextState, stateData, nextStateData) match {
         // ORDER MATTERS!
         case (WAIT_FOR_INIT_INTERNAL, OFFLINE, _, normal: DATA_NORMAL) =>
+          // LocalChannelUpdate is already published when restoring the channel
```

Comment:
I preferred having it here, it was clearer when it was explicitly done in the transition phases...

Reply:
I understand but the problem is that at this place we don't have enough information to know if the channel update changed or not.

Reply:
I've added a `channelUpdateBeforeRestore_opt` in `DATA_NORMAL` so that we have access to it in the transition function and put everything back there.